gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/task/SendACKMessageHandler.java

```diff
@@ -0,0 +1,46 @@
+package uk.nhs.adaptors.pss.translator.task;
+
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import uk.nhs.adaptors.pss.translator.mhs.MhsRequestBuilder;
+import uk.nhs.adaptors.pss.translator.mhs.model.OutboundMessage;
+import uk.nhs.adaptors.pss.translator.model.ACKMessageData;
+import uk.nhs.adaptors.pss.translator.service.ACKMessageService;
+import uk.nhs.adaptors.pss.translator.service.MhsClientService;
+
+@Slf4j
+@Component
+@AllArgsConstructor(onConstructor = @__(@Autowired))
+public class SendACKMessageHandler {
+
+    private final MhsRequestBuilder requestBuilder;
+    private final MhsClientService mhsClientService;
+    private final ACKMessageService messageService;
+
+    @SneakyThrows
+    public boolean prepareAndSendMessage(ACKMessageData messageData) {
+        String ackMessage = messageService.buildMessage(messageData);
+        OutboundMessage outboundMessage = new OutboundMessage(ackMessage);
+        var request = requestBuilder.buildSendACKRequest(
+                messageData.getConversationId(),
+                messageData.getToOdsCode(),
+                outboundMessage);
+
+        try {
+            String response = mhsClientService.send(request);
+            LOGGER.debug(response);
+        } catch (WebClientResponseException e) {
+            LOGGER.error("Received an ERROR response from MHS: [{}]", e.getMessage());
+            return false;
+        }
+
+        LOGGER.info("Got response from MHS - 202 Accepted");
```

Comment:
Can other codes be returned and should we therefore be checking the response?

Reply:
Reading the [documentation](https://github.com/nhsconnect/integration-adaptors/blob/develop/mhs/MHS-Outbound.yaml), the MHS adaptor should respond with 202 if the message is accepted and sent to spine. From my understanding, the only other outcome is an error code, for example if the MHS adaptor was down , in which case the method would return `false`. This is propagated to the initial call to `sendNackMessage`, which will also return `false`. 

If `false` is returned we could retry sending the message by recalling `sendNackMessage` from `EHRExtractRequestHandler`, otherwise in eight days the incumbent system should prompt for the patient data to be sent via envelope, as outlined in [Handling Acknowledgements](https://gpitbjss.atlassian.net/wiki/spaces/NIA/pages/11747655681/Handling+Acknowledgments) - additional scenarios.